src/tm-threads.c

```diff
@@ -1719,31 +1779,46 @@ void TmThreadClearThreadsFamily(int family)
  */
 TmEcode TmThreadSpawn(ThreadVars *tv)
 {
+    int rc;
     pthread_attr_t attr;
     if (tv->tm_func == NULL) {
         FatalError(SC_ERR_TM_THREADS_ERROR, "No thread function set");
     }
 
-    /* Initialize and set thread detached attribute */
-    pthread_attr_init(&attr);
+#ifdef HAVE_DPDK
```

Comment:
could we split this out to have 2 util funcs `TmThreadSpawnCreate` and `TmThreadSpawnCreateDPDK` or something similar?

Reply:
Did try to remove those ifdef things as much as possible here, or at least moved them to a different function. By moving it to another function I was able to mitigate the issue of wrapped `if` in ifdefs.